### ISSUE TITLE
Inventory swapping bug when holding diving suit

### ISSUE THREAD

**Issue by CurePindal** [Regalis11/Barotrauma#7218]
- [x ] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Error msg when trying to swap items that cant be swapped

**Steps To Reproduce**
Couldnt reproduce but this is how it happened:
1. Hold combat diving suit
2. Trying to swap to a stack of 2 of the assistant skill books
3. Error

**Version**
15.13, Win, Singleplayer, vanilla

**Additional information**
```
Something went wrong when trying to swap items between inventory slots: couldn't find the source item from it's inventory.
   at System.Environment.get_StackTrace()
   at Barotrauma.Inventory.TrySwapping(Int32 index, Item item, Character user, Boolean createNetworkEvent, Boolean swapWholeStack) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Inventory.cs:line 666
   at Barotrauma.Inventory.TryPutItem(Item item, Int32 i, Boolean allowSwapping, Boolean allowCombine, Character user, Boolean createNetworkEvent, Boolean ignoreCondition) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Inventory.cs:line 516
   at Barotrauma.CharacterInventory.TryPutItem(Item item, Int32 index, Boolean allowSwapping, Boolean allowCombine, Character user, Boolean createNetworkEvent, Boolean ignoreCondition) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\CharacterInventory.cs:line 443
   at Barotrauma.CharacterInventory.<>c__DisplayClass48_0.<QuickUseItem>g__Equip|0() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\CharacterInventory.cs:line 1067
   at Barotrauma.CharacterInventory.QuickUseItem(Item item, Boolean allowEquip, Boolean allowInventorySwap, Boolean allowApplyTreatment, Nullable`1 action, Boolean playSound) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\CharacterInventory.cs:line 1026
   at Barotrauma.CharacterInventory.UpdateSlotInput() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\CharacterInventory.cs:line 763
   at Barotrauma.CharacterInventory.Update(Single deltaTime, Camera cam, Boolean isSubInventory) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\CharacterInventory.cs:line 568
   at Barotrauma.CharacterHUD.Update(Single deltaTime, Character character, Camera cam) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Characters\CharacterHUD.cs:line 180
   at Barotrauma.Character.UpdateControlled(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Characters\Character.cs:line 405
   at Barotrauma.Character.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 2740
   at Barotrauma.AICharacter.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AICharacter.cs:line 38
   at Barotrauma.Character.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 2614
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 212
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 981
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 651
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 504
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 94
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 398
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 368
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 61

```

**Comment by Regalis11** [comment]
Seems to have been caused by the same issue as https://github.com/regalis11/barotrauma/issues/7153, should be fixed in https://github.com/Regalis11/Barotrauma-development/commit/342df5273f5f1fe802e390091a552deae5771e90

**Comment by Lithosagym** [comment]
Tested, no errors or crashes related to swapping a diving suit with a stack of books, appears to be fixed.

However, I did find an interesting quality about stacks of books. Having a stack of two books and then reading one of them makes the lowered percentage book appear every other time you equip/unequip the stack of books.

https://user-images.githubusercontent.com/60911734/142798037-7c6622e3-098e-4f97-bfa6-e554fdac470d.mp4
